elements/lisk-p2p/src/p2p.ts

```diff
@@ -727,6 +732,15 @@ export class P2P extends EventEmitter {
 				}
 
 				const existingPeer = this._peerPool.getPeer(peerId);
+				// Allow connections with lower version to have incoming connection even if it has outbound
+				if (
+					existingPeer &&
+					isVersionLessThan(incomingPeerInfo.version, this._nodeInfo.version)
+				) {
+					this._peerPool.addInboundPeer(incomingPeerInfo, socket);
```

Comment:
We should check that it's not already in the inbound 'duplicatePeerMap' list or else we can end up with the same peer two or more times in the  'duplicatePeerMap' and this could be a memory leak or attack vector.

Reply:
Since its setting in a map, if it finds the same key then it will overwrite it with the new value so we should be afraid of duplicates

Reply:
I am only creating the peer inbound object when there is only one connection present or none.